streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -502,7 +504,8 @@ public StreamThread(final Time time,
         this.assignmentErrorCode = assignmentErrorCode;
         this.shutdownErrorHook = shutdownErrorHook;
         this.streamsUncaughtExceptionHandler = streamsUncaughtExceptionHandler;
-        this.cacheResizer = cacheResizer;
+        this.threadCache = threadCache;
+        cacheSizes = new ConcurrentHashMap<>();
```

Comment:
Does this need to be a concurrent map? Seems to only be accessed by the StreamThread itself